services/go-filler/log-fetcher/internal/config/config.go

```diff
@@ -0,0 +1,50 @@
+package config
+
+import (
+	"log"
+	"os"
+
+	"github.com/joho/godotenv"
+)
+
+type RPCs struct {
+	ArbitrumSepolia string
+	BaseSepolia     string
+	OptimismSepolia string
+	Sepolia         string
+}
+
+type Config struct {
+	RPCs     *RPCs
+	MongoUri string
+}
+
+func NewConfig() (*Config, error) {
+	if err := godotenv.Load(".env"); err != nil {
+		return nil, err
+	}
+
+	config := &Config{
+		RPCs: &RPCs{
+			ArbitrumSepolia: getEnvStr("ARBITRUM_SEPOLIA_RPC"),
+			BaseSepolia:     getEnvStr("BASE_SEPOLIA_RPC"),
+			OptimismSepolia: getEnvStr("OPTIMISM_SEPOLIA_RPC"),
+			Sepolia:         getEnvStr("SEPOLIA_RPC"),
```

Comment:
Would recommend replacing this with a more robust flags library. There's the built in one: https://pkg.go.dev/flag. OP uses `github.com/urfave/cli/v2` heavily, which also supports envvars (e.g. https://github.com/ethereum-optimism/optimism/blob/develop/op-batcher/flags/flags.go).

Reply:
Also might consider loading a `toml` file or something, given this is per-chain config.